includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -2,18 +2,22 @@
 
 class AMP_Image_Dimension_Extractor {
 	static $callbacks_registered = false;
+	const STATUS_FAILED_LAST_ATTEMPT = 'failed';
+	const STATUS_IMAGE_EXTRACTION_FAILED = 'failed';
 
-	static public function extract( $url ) {
+	static public function extract( $urls ) {
 		if ( ! self::$callbacks_registered ) {
 			self::register_callbacks();
 		}
 
-		$url = self::normalize_url( $url );
-		if ( false === $url ) {
-			return false;
+		foreach ( $urls as &$url ) {
+			$url = self::normalize_url( $url );
```

Comment:
`normalize_url` can return `false`, which we should probably just ignore (i.e. don't pass through to extract dimensions).